index.js

```diff
@@ -1,10 +1,74 @@
 'use strict';
 const crypto = require('crypto');
 
-module.exports = length => {
+const urlSafeChars = 'abcdefjhijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789-._~'.split('');
+
+const generateForCustomChars = (length, chars) => {
+	// Generating entropy is faster than complex math operations in js, so we use the simplest way
+	const charNum = chars.length;
+	const maxValidSelector = (Math.floor(0x10000 / charNum) * charNum) - 1;
+	const entropyLen = 2 * Math.ceil(1.1 * length); // Generating a bit more than required so chances we need more than one pass will be really low
+	let entropyPos;
```

Comment:
Don't use abbreviations. 
```suggestion
	let entropyPosition;
```

`str` => `string`

etc

Reply:
done